### PR TITLE
Links to Source

### LINKED ISSUE #1218: reference page links are broken
See any link on https://coalton-lang.github.io/reference/

### PR DESCRIPTION
**Pull request by jbouwman**
Correct the generation of the local repository path under GitHub CI, so that source links are correct.

GITHUB_WORKSPACE names the directory under which Coalton is cloned: it needs `/coalton` appended.

Fixes #1218

### CODE REVIEW

File: src/doc/main.lisp [hunk 1]
```diff
@@ -19,18 +20,32 @@
...
-  "Look up the most recent commit, either in the env var GITHUB_SHA which is provided by github CI, or by calling out to git rev-parse."
+  "Get the current commit by checking the environment variable GITHUB_SHA set by GitHub CI. If it is unset, use the output of git rev-parse instead."
   (or (uiop:getenv "GITHUB_SHA")
       (string-trim '(#\Newline)
                    (uiop:run-program "git rev-parse HEAD" :output ':string))))
 
+;;;  In GitHub CI, the environment variable GITHUB_WORKSPACE names the
+;;;  directory containing the directory into which Coalton is cloned
+;;;  during build, so '/coalton' is appended to that, resulting in the
+;;;  path /__w/coalton/coalton/coalton, like summoning Beetlejuice.
```
Comment (Izaakwltn) [inline src/doc/main.lisp:35]: Betelgeuse, Betelgeuse, Betelgeuse, but I'm not sure if humor is permissible in comments.

File: src/doc/main.lisp [hunk 2]
```diff
@@ -19,18 +20,32 @@
...
+
 (defun local-path ()
-  (or (uiop:getenv "GITHUB_WORKSPACE")
-      (namestring (asdf:system-source-directory "coalton"))))
+  "Return the path to the directory containing Coalton source."
+  (let ((ci-workspace (uiop:getenv "GITHUB_WORKSPACE")))
+    (cond (ci-workspace
```
Comment (Izaakwltn) [inline src/doc/main.lisp:40]: is there a reason to be a bifurcated`cond` instead of `if`?

**Review by Izaakwltn — APPROVED** [review]
Looks good, `yes, tiny nits, ey?`

File: src/doc/main.lisp (hunk 1 above)
Comment (jbouwman) [reply src/doc/main.lisp:35]: Then why is it allowed in docstrings!

https://github.com/coalton-lang/coalton/blob/6f0753b0a548ce40ef6f05d5c5ab0287226408f3/library/hashtable.lisp#L71

File: src/doc/main.lisp (hunk 2 above)
Comment (jbouwman) [reply src/doc/main.lisp]: converted to if

File: src/doc/main.lisp (hunk 1 above)
Comment (jbouwman) [reply src/doc/main.lisp]: levity eradicated

**Pull request merged**